### PR TITLE
Validate workspace-databases.json

### PR DESCRIPTION
**Pull request by norascheuch**
We want to add a json schema file to validate the new workspace-databases file. 

## Checklist

- [ ] [CHANGELOG.md](https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/CHANGELOG.md) has been updated to incorporate all user visible changes made by this pull request.
- [ ] Issues have been created for any UI or other user-facing changes made by this pull request.
- [ ] _[Maintainers only]_ If this pull request makes user-facing changes that require documentation changes, open a corresponding docs pull request in the [github/codeql](https://github.com/github/codeql/tree/main/docs/codeql/codeql-for-visual-studio-code) repo and add the `ready-for-doc-review` label there.

### CODE REVIEW

**Review by charisk — COMMENTED** [review]
This is looking good. However (please don't hate me 🙀), @koesie10 has mentioned https://code.visualstudio.com/api/references/contribution-points#contributes.jsonValidation to me which we could take advantage of instead of having to append `"schema": "file://..."`. Do you want to try that out and see how it goes?

File: extensions/ql-vscode/src/extension.ts [hunk 1]
```diff
@@ -1225,6 +1226,11 @@ async function activateWithInstalledDistribution(
     )
   );
 
+  const storagePath = ctx.storageUri?.fsPath || ctx.globalStorageUri.fsPath;
```
[the remark this answers is not in the document]
Comment (norascheuch) [reply extensions/ql-vscode/src/extension.ts:1229]: 👍

File: extensions/ql-vscode/src/databases/db-config-store.ts [hunk 2]
```diff
@@ -34,7 +35,13 @@ export class DbConfigStore extends DisposableObject {
 
   private async loadConfig(): Promise<void> {
     if (!await fs.pathExists(this.configPath)) {
-      await fs.writeJSON(this.configPath, this.createEmptyConfig(), { spaces: 2 });
+      const schemaPath = path.resolve(this.extensionPath, 'workspace-databases-schema.json');
+      await fs.writeJSON(this.configPath, fs.readJson(schemaPath), {});
```
[the remark this answers is not in the document]
Comment (norascheuch) [reply extensions/ql-vscode/src/databases/db-config-store.ts:39]: indeed!

File: extensions/ql-vscode/test/pure-tests/databases/data/workspace-databases-schema.json [hunk 3]
```diff
@@ -0,0 +1,49 @@
+{
```
[the remark this answers is not in the document]
Comment (norascheuch) [reply extensions/ql-vscode/test/pure-tests/databases/data/workspace-databases-schema.json:1]: Right now we're not testing the validation and are not returning an error if the schema doesn't exist. So it doesn't matter at all, I added it for completeness sake. I'm not sure if we can access the extension context in the pure test to get the original schema file though?

**Review by charisk — APPROVED** [review]
LGTM!

**Pull request merged**